pkg/manager/impl/artifact_manager.go

```diff
@@ -137,65 +144,83 @@ func (m *artifactManager) GetArtifact(ctx context.Context, request *datacatalog.
 
 	datasetID := request.Dataset
 
+	artifactModel, err := m.findArtifact(ctx, datasetID, request)
+	if err != nil {
+		logger.Errorf(ctx, "Failed to retrieve artifact for get artifact request %v, err: %v", request, err)
+		m.systemMetrics.getFailureCounter.Inc(ctx)
+		return nil, err
+	}
+
+	artifact, err := transformers.FromArtifactModel(artifactModel)
+	if err != nil {
+		logger.Errorf(ctx, "Error in transforming get artifact request %+v, err %v", artifactModel, err)
+		m.systemMetrics.transformerErrorCounter.Inc(ctx)
+		return nil, err
+	}
+
+	artifactDataList, err := m.getArtifactDataList(ctx, artifactModel.ArtifactData)
+	if err != nil {
+		m.systemMetrics.getFailureCounter.Inc(ctx)
+		return nil, err
+	}
+	artifact.Data = artifactDataList
+
+	logger.Debugf(ctx, "Retrieved artifact dataset %v, id: %v", artifact.Dataset, artifact.Id)
+	m.systemMetrics.getSuccessCounter.Inc(ctx)
+	return &datacatalog.GetArtifactResponse{
+		Artifact: artifact,
+	}, nil
+}
+
+type artifactQueryHandle interface {
+	GetArtifactId() string
+	GetTagName() string
+}
+
+func (m *artifactManager) findArtifact(ctx context.Context, datasetID *datacatalog.DatasetID, queryHandle artifactQueryHandle) (models.Artifact, error) {
 	var artifactModel models.Artifact
-	switch request.QueryHandle.(type) {
-	case *datacatalog.GetArtifactRequest_ArtifactId:
-		logger.Debugf(ctx, "Get artifact by id %v", request.GetArtifactId())
-		artifactKey := transformers.ToArtifactKey(datasetID, request.GetArtifactId())
+
+	key := queryHandle.GetArtifactId()
```

Comment:
Is there any reason we changed this from using a switch over to `len(queryHandle.GetArtifcatId())` from `request.QueryHandle.(type)`. Was it required because of the addition of UpdateArtifactRequest handling?

Reply:
@hamersaw Yes, exactly. Unfortunately the type generated for the `UpdateArtifactRequest` is a `datacatalog.UpdateArtifactRequest_ArtifactId` (instead of a `datacatalog.GetArtifactRequest_ArtifactId`), however both implement `GetArtifactId() string`.
As the `oneof` interface is not exported in the generated protobuf definitions, we can't use a shared interface, so this was the easiest I came up with (which allows both to be passed as a single method parameter).